core_lib/src/canvaspainter.cpp

```diff
@@ -39,7 +38,20 @@ CanvasPainter::~CanvasPainter()
 void CanvasPainter::setCanvas(QPixmap* canvas)
 {
     Q_ASSERT(canvas);
-    mCanvas = canvas;
+    if (mCanvas == nullptr || mCanvasSize != canvas->size()) {
+
+        mCanvas = canvas;
```

Comment:
It is yeah, the reason is that we always call setCanvas but the canvas might not have been modified, so to avoid unnecessarily filling those pixmaps on each call, we check first if there's actually a need for it.

Although before it might not have made much sense to check for mCanvas == nullptr, after your cleanup though it does, to avoid accessing a potential nullptr

Reply:
You made some great changes. You're absolutely right that the interface didn't match the way it was used. Having the painter be initialized with a pixmap and resetting it upon resizing is an excellent simplification.

Thank you